### ISSUE TITLE
Calendar is working good in iOS but for android it has bug in initial month view.

### ISSUE THREAD

**Issue by devekoperiOS** [christopherdro/react-native-calendar#43]
I am trying to use calendar in iOS and android apps. But in android app, its seems to be error in initial month. Its setting date form wrong day. If I click on next or prev and then come back to initial screen, then it fixes. What might be missing in setting up initial month view?

**Comment by devekoperiOS** [comment]
I have removed "scrollEnabled={true}" from calendar tag and it worked. I wrote the following.
<Calendar
          ref="calendar"
          scrollEnabled={true}
          showControls={true}
          dayHeadings={customDayHeadings}
          titleFormat={'MMMM YYYY'}
          prevButtonText={'Prev'}
          nextButtonText={'Next'}
          onDateSelect={(date) => this.setState({selectedDate: date})}
          onTouchPrev={() => console.log('Back TOUCH')}
          onTouchNext={() => console.log('Forward TOUCH')}
          onSwipePrev={() => console.log('Back SWIPE')}
          onSwipeNext={() => console.log('Forward SWIPE')}/>

**Comment by vincentfretin** [comment]
I saw the same issue yesterday. Please reopen, this is a workaround, not a fix.
I still have the issue with RN 0.28 on android. We are in june and on the first load, it goes to april but with the title of june. And indeed, if scrollEnabled is false, the issue goes away.
I think the issue may be in scrollEnded function. If somebody want to look at it.

**Comment by vincentfretin** [comment]
Another issue I found on android is that when I swipe to go to next month, it decrements the month instead of incrementing it. I think it may be related.

**Comment by mCodex** [comment]
@devekoperiOS I realize the same thing yesterday.

**Comment by devekoperiOS** [comment]
Temporary solution can be disabling scrolling.

**Comment by mCodex** [comment]
Yes, thank you!

**Comment by KGALLET** [comment]
It seems that i've got the same issue on iOS real device.
Disabling scrolling temporary fixed it.

**Comment by nodegin** [comment]
removing `scrollEnabled={true}` fixed this issue, any solution?

**Comment by nodegin** [comment]
In addition I found that by adding `showControls={true}` the `titleFormat` will be broken too.

Without `showControls` the title shows `August 2016`, but with `showControls` the title shows `Aug 2016`

**Comment by MrToph** [comment]
I also encountered the problem and figured out how to use it with _scrollEnabled_ if anyone's interested.

The way it works in the code is that the ScrollView has 5 elements pre-rendered, from [month-2, ..., month+2]. So you have to call `this.scrollToItem(2)` for it to scroll to index 2, the current month.
When the Calendar mounts, it tries to scroll in `componentDidMount()`, however you _cannot_ scroll in componentDidMount as at this point some animations will still be run ([see here](http://stackoverflow.com/questions/33208477/react-native-android-scrollview-scrollto-not-working)).
So the solution is to call `this.scrollToItem(VIEW_INDEX)` _after_ running the animations.
**TL;DR**
Change the line in Calendar.js

``` javascript
componentDidMount() {
      this.scrollToItem(VIEW_INDEX)
}
```

to

``` javascript
import {...,
  InteractionManager
} from 'react-native';
componentDidMount() {
     InteractionManager.runAfterInteractions(() => {
      this.scrollToItem(VIEW_INDEX)
    })
}
```

I might create a pull request later.

**Comment by nicotroia** [comment]
I had a similar issue where the Calendar would not render when scrollEnabled was set to `false`. 

The View container that wraps the calendar days needs to have a style change in order to work.

````
{this.props.scrollEnabled ? (
  <ScrollView
    ...
  >
    {calendarDates.map((date) => this.renderCalendarView(this.props.calendarFormat, moment(date), eventDatesMap))}
  </ScrollView>
 ) : (
  <View style={{flex: 1}} ref={calendar => this._calendar = calendar}>
    {calendarDates.map((date) => this.renderCalendarView(this.props.calendarFormat, moment(date), eventDatesMap))}
  </View>
)}
````

I added the `style={{flex: 1}}` to the <View> container (when there is no ScrollView) and it fixed my problem. Hopefully this will help someone else
